### PR TITLE
Overhaul Guides and Walkthroughs

### LINKED ISSUE #132: Remove references to d.ims.io, slack etc.
There are some references to d.ims.io and slack in the documentation and release notes. Make sure you strip all references and replace them with generic DNS names. [Example.com](https://en.wikipedia.org/wiki/Example.com) has been reserved for this sort of purpose.

### LINKED ISSUE #169: Overhaul Walkthroughs and Guides
Our guides and walkthroughs have fallen out of sync, so here's an overarching plan to realign them.

**The Guides**
- [x] An Iterative Walkthrough
   - [x] Guestbook (#170)
   - [x] Guestbook + Redis (#171)
   - [x] Guestbook + Redis + Consul (#172)
- [x] Terraform Beyond Layer0 (#173)
- [x] One-Off Tasks (#174)

### PR DESCRIPTION
**Pull request by tlake**
**Important Note: Once this is approved but before it's merged in, we need to generate the actual documentation with `make build` in `docs-src/` and then commit those changes to this PR.**

---

This work presents a retooled walkthrough and two other guides that dive deeper into certain aspects of Layer0 and Terraform.

### The Walkthrough

The walkthrough is composed of an introduction and three iterative sections of example systems. Each section contains parallel, step-by-step instructions on how to deploy an increasingly-complex system using either the Layer0 CLI directly or using Terraform and the Layer0 plugin.

- Deployment 1: Guestbook
- Deployment 2: Redis + Guestbook
- Deployment 3: Consul + Redis + Guestbook

### One-Off Tasks

This guide exists as an example of using Layer0 to spin up a short-running task that needs to run once and then stop.

### Terraform Beyond Layer0

This guide exists to demonstrate how a user could use Terraform to deploy a system that uses both Layer0 concepts and concepts outside of Layer0's purview.

---

Not a part of this PR, but this work also resulted in the creation of a Redis module which we maintain as a Layer1 service (located [here](https://github.com/quintilesims/redis)) similar to how we maintain a Consul module as a Layer1 service. Additionally, the name of the repository which holds the task definitions, terraform files, and other resources used in the walkthrough has been changed from `layer0-examples` to `guides` (located [here](https://github.com/quintilesims/guides)) to better sync with our documentation naming conventions.

---

This also includes a bit of work here and across a couple of our Layer1 repos to remove references to d.ims.io and our internal Slack in public-facing docs.

---

Closes #169
Closes #132

### CODE REVIEW

File: docs-src/docs/guides/walkthrough/intro.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# An Iterative Walkthrough
+
+This guide aims to take you through three increasingly-complex deployment examples using Layer0.
+Successive sections build upon the previous ones, and each deployment can be completed either through the Layer0 CLI directly, or through Terraform using our custom [Layer0 Terraform Provider](/reference/terraform-plugin).
```
Comment (diemonster) [inline docs-src/docs/guides/walkthrough/intro.md]: I believe this is the first time we mention terraform in our docs, if following them linearly. I'm thinking we should add a hyperlink here to terraform's docs and add some details for terraform in a section like `/setup/install/`

Comment (tlake) [inline docs-src/docs/guides/walkthrough/intro.md]: Easy enough to add that link and a bit of clarification here. As for more details, we do cover a little bit in that `/reference/terraform-plugin` link, which includes a couple of links to Terraform docs and how to install Terraform. Think we need more? Do you want to have a separate page that talks about installing terraform?

Comment (diemonster) [inline docs-src/docs/guides/walkthrough/intro.md]: I think we just need something that serves as a proper introduction to `terraform`.. why do we use it? What is it good for? What does it compare to? Etc... 

Such a file could exist in the `references` section and be linked from somewhere in the introduction. And yes, we do cover some of this in `reference` already, but I feel like the existing `/reference/terraform-plugin` section is more about our Layer0 provider than it is about terraform specifically.

Comment (diemonster) [inline docs-src/docs/guides/walkthrough/intro.md]: We can spin this off into a separate issue https://github.com/quintilesims/layer0/issues/251

**Pull request merged**